service/src/test/java/org/sunbird/learner/actors/skill/UserSkillManagementActorTest.java

```diff
@@ -0,0 +1,373 @@
+    innerMap.put(JsonKey.REQUESTED_BY, userId);
+    actorMessage.setContext(innerMap);
+    actorMessage.put(JsonKey.ENDORSED_USER_ID, endorseUserId);
+    actorMessage.setOperation(ActorOperations.GET_SKILL.getValue());
+    return actorMessage;
+  }
+
+  @Test
+  public void testAddSkill() {
+    Mockito.reset(cassandraOperation);
+    addSkill();
+    Mockito.reset(cassandraOperation);
+  }
+
+  private void addSkill() {
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+
+    Response response = createCassandraInsertSuccessResponse();
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(response);
+    Response response1 = createGetUserSuccessResponse();
+    when(cassandraOperation.getRecordById(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(response1);
+    subject.tell(getAddSkillRequest(USER_ID, ENDORSED_USER_ID, skillsList), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res);
+  }
+
+  @Test
+  public void testAddSkillAgain() {
+    Mockito.reset(cassandraOperation);
+    addSkill();
+    Mockito.reset(cassandraOperation);
+  }
+
+  private Response createCassandraInsertSuccessResponse() {
+    Response response = new Response();
+    response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
+    return response;
+  }
+
+  private Response createGetUserSuccessResponse() {
+    Response response = new Response();
+    Map<String, Object> userMap = new HashMap<>();
+    userMap.put(JsonKey.ID, USER_ID);
+    userMap.put(JsonKey.ROOT_ORG_ID, ROOT_ORG_ID);
+    List<Map<String, Object>> result = new ArrayList<>();
+    result.add(userMap);
+    response.put(JsonKey.RESPONSE, result);
+    return response;
+  }
+
+  private Response createGetUserFailureResponse() {
+    Response response = new Response();
+    List<Map<String, Object>> result = new ArrayList<>();
+    response.put(JsonKey.RESPONSE, result);
+    return response;
+  }
+
+  private Response createGetSkillsSuccessResponse() {
+    Response response = new Response();
+    Map<String, Object> userMap = new HashMap<>();
+    List<Map<String, Object>> result = new ArrayList<>();
+    result.add(userMap);
+    response.put(JsonKey.RESPONSE, result);
+    return response;
+  }
+
+  @Test
+  public void testUpdateSkill() {
+    Mockito.reset(cassandraOperation);
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+
+    Request actorMessage = getUpdateSkillRequest(USER_ID, skillsList);
+    HashMap<String, Object> esDtoMap = new HashMap<>();
+    when(cassandraOperation.getRecordById(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(createGetUserSuccessResponse());
+
+    Map<String, Object> filters = new HashMap<>();
```

Comment:
Create a helper method (e.g. mockGetSkillResponse) to mock get user skill response.